synapse/push/mailer.py

```diff
@@ -399,6 +394,16 @@ async def _get_room_vars(
 
         return room_vars
 
+    async def _get_room_avatar(
+        self,
+        room_state_ids: StateMap[str],
+    ) -> Optional[str]:
+        event_id = room_state_ids.get((EventTypes.RoomAvatar, ""))
+        if event_id:
+            ev = await self.store.get_event(event_id)
+            return ev.content["url"]
```

Comment:
This is untrusted, so there's no guarantee that this will have a `url` (or that it will be a string).

I also wonder if it makes sense to use the `thumbnail_url` instead?

Regardless, we'll need to convert the MXC URL to an HTTP one, I believe.

Reply:
> Regardless, we'll need to convert the MXC URL to an HTTP one, I believe.

I think this is already done in the jinja template room.html:
```jinja2
                <img alt="" src="{{ room.avatar_url|mxc_to_http(48,48) }}" />
```
where I'm guessing that pipe will feed the LHS into the `mxc_to_http_filter` defined in synapse/util/templates.py:85

Reply:
> This is untrusted, so there's no guarantee that this will have a url (or that it will be a string).

The spec says that "url" is required---but I guess we don't enforce that at present? Can do an `isintance(..., str)` check on the URL though.

> I also wonder if it makes sense to use the thumbnail_url instead?

Spec says that this is "Only present if the thumbnail is unencrypted.". But I'm not sure if the main avatar could be encrypted too?

Reply:
> > This is untrusted, so there's no guarantee that this will have a url (or that it will be a string).
> 
> The spec says that "url" is required---but I guess we don't enforce that at present? Can do an `isintance(..., str)` check on the URL though.

In general, event content can be whatever. So I don't think we can assume it is there / the right type. Even if we enforce it, an event could come over federation that has the incorrect shape. (Or it could have been redacted so all event content was removed.)